src/web/assets/garnish/src/MenuBtn.js

```diff
@@ -63,7 +65,21 @@ export default Base.extend(
       this.addListener(this.$btn, 'mousedown', 'onMouseDown');
       this.addListener(this.$btn, 'keydown', 'onKeyDown');
       this.addListener(this.$btn, 'blur', 'onBlur');
-      this.enable();
+
+      this.observer = new MutationObserver((mutations) => {
+        mutations.forEach((mutation) => {
+          if (
+            mutation.type === 'attributes' &&
+            mutation.attributeName === 'disabled'
+          ) {
+            this.handleStatusChange();
+          }
+        });
```

Comment:
Could this be replaced with a `for...of` loop so you can `break` out when you find the attribute you’re looking for?

Reply:
Updated.